pkg/bundle/service.go

```diff
@@ -42,92 +42,91 @@ func getConfigMapObjectKey(cluster *clusterv1.Cluster) types.NamespacedName {
 	}
 }
 
-// Configure configures the observability-bundle application.
-// the observabilitybundle application to enable monitoring agents.
+// Configure creates or updates the observability-bundle configuration based on
+// cluster feature flags and links it to the bundle app via extraConfigs.
 func (s BundleConfigurationService) Configure(ctx context.Context, cluster *clusterv1.Cluster) error {
-	logger := log.FromContext(ctx).WithValues("appName", observabilityBundleAppName)
-	logger.Info("configuring application")
+	logger := log.FromContext(ctx)
+	logger.Info("configuring bundle")
 
-	bundleConfiguration := bundleConfiguration{
-		Apps: map[string]app{
-			commonmonitoring.MonitoringAlloyAppName: {
-				AppName: commonmonitoring.AlloyMonitoringAgentAppName,
+	bundleConfig := s.buildBundleConfiguration(cluster)
 
-				Enabled: s.config.Monitoring.IsMonitored(cluster),
-			},
-		},
-	}
-	logger.Info("create or update configmap")
-	err := s.createOrUpdateObservabilityBundleConfigMap(ctx, cluster, bundleConfiguration)
+	logger.Info("creating or updating configmap")
+	err := s.createOrUpdateConfigMap(ctx, cluster, bundleConfig)
 	if err != nil {
-		return fmt.Errorf("failed to create or update observability bundle configmap: %w", err)
+		return fmt.Errorf("failed to create/update bundle configmap: %w", err)
 	}
+	logger.Info("configmap created or updated successfully")
 
-	logger.Info("configure application")
-	err = s.configureObservabilityBundleApp(ctx, cluster)
+	logger.Info("configuring bundle")
+	err = s.configureApp(ctx, cluster)
 	if err != nil {
-		return fmt.Errorf("failed to configure observability bundle app: %w", err)
+		return fmt.Errorf("failed to configure bundle: %w", err)
 	}
 
-	logger.Info("application is configured successfully")
+	logger.Info("bundle configured successfully")
 
 	return nil
 }
 
-func (s BundleConfigurationService) createOrUpdateObservabilityBundleConfigMap(
-	ctx context.Context, cluster *clusterv1.Cluster, configuration bundleConfiguration) error {
-
-	logger := log.FromContext(ctx)
-
-	values, err := yaml.Marshal(configuration)
-	if err != nil {
-		return fmt.Errorf("failed to marshal configuration to yaml: %w", err)
+// buildBundleConfiguration creates the bundle configuration based on cluster feature flags.
+func (s BundleConfigurationService) buildBundleConfiguration(cluster *clusterv1.Cluster) bundleConfiguration {
+	return bundleConfiguration{
+		Apps: map[string]app{
+			apps.AlloyMetricsHelmValueKey: {
+				AppName: apps.AlloyMetricsAppName,
+				Enabled: s.config.Monitoring.IsMonitoringEnabled(cluster),
+			},
+			apps.AlloyLogsHelmValueKey: {
+				AppName: apps.AlloyLogsAppName,
+				Enabled: s.config.Logging.IsLoggingEnabled(cluster),
+			},
+			apps.AlloyEventsHelmValueKey: {
+				AppName: apps.AlloyEventsAppName,
+				Enabled: s.isEventsEnabled(cluster),
+			},
```

Comment:
I think we are adding 2 new entries in this configuration which were not there before. Why is this ?